csharp/src/Apache.Arrow/BitUtility.cs

```diff
@@ -62,6 +62,64 @@ public static void SetBit(Span<byte> data, int index, bool value)
                 : (byte)(data[idx] & ~BitMask[mod]);
         }
 
+        /// <summary>
+        /// Set the number of bits in a span of bytes starting
+        /// at a specific index, and limiting to length.
+        /// </summary>
+        /// <param name="data">Span to set bits value.</param>
+        /// <param name="offset">Bit index to start counting from.</param>
+        /// <param name="length">Maximum of bits in the span to consider.</param>
+        public static void SetBits(Span<byte> data, int index, int length, bool value)
```

Comment:
Since this is a public API, can we do some argument checking? For example, `index + length < data.Length` and both `index` and `length` are non-negative.

Reply:
You are right. However, after taking a closer look at BitUtility class I have some concern that BitUtility should be a public API. And here are reasons:
1) There aren't any argument checking in all other methods (however, there are debug.asserts - I think it's done due to perfomance reason)
2) This class is implemented with an assumption of being used with Bitmap only, when we know that the number of meaningful bits should be less than Max.Int (because each bit corresponds to the validity of value in Value.Span, wich length is limit by Int). In general case the amount of bit inside a Span<byte> can be Max,Int * 8 and correct methods for working with bits inside a span should have long value for bit index and for length.

I added required checks into SetBits and CountBits methods (as this two are very similar) and didn't change other methods. You may see the resulting code in PR, but actually it looks intermediate compromise between two solutions:
 
change all BitUtility methods to use longs for bit indices (with possible performance drawdown)

or

make BitUtility internal (breaking change, in case somebody uses BitUtility in their code)

Would you please make some inputs on that?

Reply:
I think a decent compromise for this PR is to not add a new public API to BitUtility. Instead, make the new method `internal`. Then you don't need to add argument checking to this method.

Then log an issue proposing making the whole `BitUtility` class `internal`. We should make that decision separate of this PR.

Reply:
Fixed